ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/forms/CollectionSelection.tsx

```diff
@@ -0,0 +1,207 @@
+import React, { useMemo, useState, ReactElement, useCallback, useEffect } from 'react';
```

Comment:
@dvail since I think you made the component. Let me know if this approach is good with you. My thought process was to make a copy and make slight changes to work with the new feature. When we remove the old reporting feature, we'll remove that original component. All I really changed was the following:

1. Changed `scopeId` to `selectedScopeID`
2. Changed `setFieldValue` to `onChange` to work with the how the new version updates the form values
3. I added a `useEffect` that triggers when `data` changes. I wanted to make sure if the user selects a scope, goes to the collections page, deletes that scope, and returns to form again, the field would reset. Apparently changing routes doesn't unmount the component so the form values remain

Reply:
Yeah this makes sense to me, no point in trying to abstract this into a single component if we are deleting the old one in the very near future. 👍 

Point (3) is a good idea, but I'm confused as to why it is necessary. Intuitively I would think that the component would get unmounted and the state would clear if you navigate to a new page. Other forms in the app have this behavior. Unless we have some kind of weird router-memory leak that keeps the component around forever, I would guess that the `const default` [here](https://github.com/stackrox/stackrox/pull/6705/files#diff-08410c0ca27de915fc74dd70b04061e9316aa98e6abb843cd83b92efe85d4a9aR32) is getting contaminated in some way, causing the values to stick around between mount/unmount lifecycles.